specs/interop/storage-upgrade.md

```diff
@@ -38,7 +38,11 @@ and add the following functionality:
 bytes32 public hashOnion;
 
 function setHashOnion(bytes32 _hashOnion) external {
-    require(msg.sender == Predeploys.PROXY_ADMIN, "Unauthorized");
+    require(msg.sender == Predeploys.L2_CROSS_DOMAIN_MESSENGER, "Unauthorized");
+
+    require((Predeploys.L2_CROSS_DOMAIN_MESSENGER).xDomainMessageSender() ==
+        AddressAliasHelper.undoL1ToL2Alias((Predeploys.PROXY_ADMIN).owner()), "Unauthorized");
```

Comment:
I think this needs to be the inverde, the alias undo should be done over the L2 Messenger, whereas the cross domain message sender remains the same